src/WebJobs.Extensions.DurableTask/LinuxAppServiceFileLogger.cs

```diff
@@ -173,9 +170,23 @@ private void RollFiles()
             // Rename current file to older file.
 
 #if !FUNCTIONS_V1
-            Syscall.rename(this.logFilePath, this.archiveFilePath);
+            try
+            {
+                rename(this.logFilePath, this.archiveFilePath);
+            }
+            catch (Exception ex)
+            {
+                throw new Exception("Linux Logger failed to roll files", ex);
```

Comment:
I don't think catching the exception at this level is helpful, it is just obfuscating whatever exception is causing the issue.